src/transformers/models/mask2former/configuration_mask2former.py

```diff
@@ -0,0 +1,334 @@
+            description
+        is_train (`boolean``, *optional*, defaults to False):
+            description
+        use_auxiliary_loss (`boolean``, *optional*, defaults to True):
+            If `True` [`Mask2FormerForUniversalSegmentationOutput`] will contain the auxiliary losses computed using
+            the logits from each decoder's stage.
+        feature_strides (`List[int]`, *optional*, defaults to [4, 8, 16, 32]):
+            Feature strides corresponding to features generated from backbone network
+        deep_supervision (`boolean``, *optional*, defaults to True):
+            description
+
+    Raises:
+        `ValueError`:
+            Raised if the backbone model type selected is not in `["swin"]`.
+
+    Examples:
+
+    ```python
+    >>> from transformers import Mask2FormerConfig, Mask2FormerModel
+
+    >>> # Initializing a Mask2Former facebook/mask2former-instance-swin-small-coco configuration
+    >>> configuration = Mask2FormerConfig()
+
+    >>> # Initializing a model (with random weights) from the facebook/mask2former-instance-swin-small-coco style configuration
+    >>> model = Mask2FormerModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```
+
+    """
+    model_type = "mask2former"
+    backbones_supported = ["swin"]
+    attribute_map = {"hidden_size": "mask_feature_size"}
+
+    def __init__(
+        self,
+        backbone_config: Optional[Dict] = None,
+        decoder_config: Optional[Dict] = None,
+        ignore_value=255,
+        num_queries=150,
+        no_object_weight=0.1,
+        class_weight=2.0,
+        mask_weight=5.0,
+        dice_weight=5.0,
+        train_num_points=12544,
+        oversample_ratio=3.0,
+        importance_sample_ratio=0.75,
+        init_std=0.02,
+        init_xavier_std=1.0,
+        layer_norm_eps=1e-05,
+        is_train=False,
+        use_auxiliary_loss=True,
+        feature_strides=[4, 8, 16, 32],
+        deep_supervision=True,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        if backbone_config is None:
+            backbone_config = SwinConfig(
+                image_size=224,
+                in_channels=3,
+                patch_size=4,
+                embed_dim=96,
+                depths=[2, 2, 6, 2],
+                num_heads=[3, 6, 12, 24],
+                window_size=7,
+                drop_path_rate=0.3,
+                out_features=["stage1", "stage2", "stage3", "stage4"],
+            )
+        else:
+            backbone_model_type = (
+                backbone_config.pop("model_type") if isinstance(backbone_config, dict) else backbone_config.model_type
+            )
+            if backbone_model_type not in self.backbones_supported:
+                raise ValueError(
+                    f"Backbone {backbone_model_type} not supported, please use one of"
+                    f" {','.join(self.backbones_supported)}"
+                )
```

Comment:
This can be removed, see my comment above

(we can do the same for MaskFormer)